libs/dao/src/main/java/com/akto/dao/SingleTypeInfoDao.java

```diff
@@ -677,22 +679,67 @@ public long fetchEndpointsCount(int startTimestamp, int endTimestamp, Set<Intege
 
             pipeline.add(Aggregates.match(Filters.in("apiCollectionId", nonHostApiCollectionIds)));
 
-            BasicDBObject groupedId = 
+            BasicDBObject groupedId =
                 new BasicDBObject("apiCollectionId", "$apiCollectionId")
                 .append("url", "$url")
                 .append("method", "$method");
             pipeline.add(Aggregates.group(groupedId, Accumulators.min("startTs", "$timestamp")));
             pipeline.add(Aggregates.match(Filters.gte("startTs", startTimestamp)));
             pipeline.add(Aggregates.match(Filters.lte("startTs", endTimestamp)));
             pipeline.add(Aggregates.sort(Sorts.descending("startTs")));
+
+            long daysBetween = (endTimestamp - startTimestamp) / (24 * 3600);
+
+            if (daysBetween <= 30) {
+                Bson addFieldsStage = Aggregates.addFields(
+                        new Field<>("day", new BasicDBObject("$dateToString",
+                                new BasicDBObject("format", "%Y-%m-%d")
+                                        .append("date", new BasicDBObject("$toDate",
+                                                new BasicDBObject("$multiply", Arrays.asList(
+                                                        new BasicDBObject("$toLong", "$timestamp"), 1000
+                                                ))
+                                        ))
+                        ))
+                );
+                Bson groupStage = Aggregates.group("$day", Accumulators.sum("count", 1));
+                Bson sortStage = Aggregates.sort(new BasicDBObject("_id", 1));
+                pipeline.add(addFieldsStage);
+                pipeline.add(groupStage);
+                pipeline.add(sortStage);
+            } else if (daysBetween <= 210) {
+                addFieldsGroupAndSort(pipeline, "week");
+            } else {
+                addFieldsGroupAndSort(pipeline, "month");
+            }
+
             MongoCursor<BasicDBObject> endpointsCursor = SingleTypeInfoDao.instance.getMCollection().aggregate(pipeline, BasicDBObject.class).cursor();
             while(endpointsCursor.hasNext()) {
-                count += 1;
-                endpointsCursor.next();
+                BasicDBObject doc = endpointsCursor.next();
+                if(doc.isEmpty()) continue;
```

Comment:
this check is fine, but can doc be empty here ?

Reply:
ummm I guess no...